stages/org.osbuild.chmod

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/python3
+"""
+Change file mode bits
+
+Change the file mode bits of one or more files or directories inside the tree.
+"""
+
+import subprocess
+import sys
+import os
+
+import osbuild.api
+
+SCHEMA = """
+"additionalProperties": false,
+"required": ["paths", "mode"],
+"properties": {
+  "paths": {
+    "type": "array",
+    "description": "Paths to operate on for changing mode bits",
+    "items": {
+      "type": "string"
+    }
+  },
+  "mode": {
+    "type": "string",
+    "description": "Symbolic or numeric octal mode"
+  },
+  "recursive": {
+    "type": "boolean"
+  }
+}
+
+"""
+
+def main(tree, options):
+    paths = options["paths"]
+    mode = options["mode"]
+    recursive = options.get("recursive", False)
+
+    paths = [os.path.join(tree, p.lstrip("/")) for p in paths]
```

Comment:
Maybe `os.relpath(p, start="/")`? Also, maybe we should refuse any non-absolute paths?

Reply:
That would be safer.  Would even save us from things like `//some/weird/path` or `../../../risky/bizniz`.

As for disallowing non-abs paths, that could also make sense.  We could do it in the schema and fail early.

Reply:
I have been using `lstrip("/")` in a few places, I think it is an ok compromise of supporting absolute paths in the options, which feel more natural when describing a root file system, while also not trying to guard against funny paths. I mean if you want you can always shoot yourself in the foot with a low level tool like osbuild. NB: `"//foo/bar/se".lstrip("/")` does the right thing and return `'foo/bar/se'`.
`os.path.relpath()` should work fine too, but only if `p` really is an absolute path, otherwise it will call `abspath` and which prepend the current working directory [Lib/posixpath.py#L464](https://github.com/python/cpython/blob/baecfbd849dbf42360d3a84af6cc13160838f24d/Lib/posixpath.py#L464).

Reply:
> NB: `"//foo/bar/se".lstrip("/")` does the right thing and return `'foo/bar/se'`.

ah, good point.  Forgot it strips all leading instances.

> `os.path.relpath()` should work fine too, but only if `p` really is an absolute path, otherwise it will call `abspath` which will prepend the current working directory [Lib/posixpath.py#L464](https://github.com/python/cpython/blob/baecfbd849dbf42360d3a84af6cc13160838f24d/Lib/posixpath.py#L464).

So do you think it's worth restricting to abs paths only?

Reply:
I think I am not a fan of `os.path.relpath` since it can call `abspath`, which brings in `curdir`. I have looked a bit more closely it its source and the path normalisation is done internally with `normpath` via `abspath`. **If** we want to be more clever than just `lstrip("/")` then maybe something like:
```
def treepath(path, tree):
   path = "/" + os.fspath(path).    # make sure path is absolute, might create //foo/bar, or /foo/bar/ or /../../foo/../bar/
   path = os.path.normpath(path)    # will remove ../../.. and //, but leave the path absolute
   relpath = path.lstrip("/")       # path is now relative
   return os.path.join(tree, path)  # absolute path rooted at tree
```
Have that as an utility function that can be used throughout the code-base. Opinions?

Reply:
*sigh* Why would they creep in `curdir` everywhere... I think I prefer `lstrip()` in this case.